core/src/main/java/com/sequenceiq/cloudbreak/rotation/executor/CMUserRotationExecutor.java

```diff
@@ -0,0 +1,107 @@
+package com.sequenceiq.cloudbreak.rotation.executor;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cluster.api.ClusterSecurityService;
+import com.sequenceiq.cloudbreak.dto.StackDto;
+import com.sequenceiq.cloudbreak.rotation.context.CMUserRotationContext;
+import com.sequenceiq.cloudbreak.rotation.secret.RotationContext;
+import com.sequenceiq.cloudbreak.rotation.secret.RotationExecutor;
+import com.sequenceiq.cloudbreak.rotation.secret.SecretLocationType;
+import com.sequenceiq.cloudbreak.rotation.secret.SecretRotationException;
+import com.sequenceiq.cloudbreak.service.CloudbreakException;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterApiConnectors;
+import com.sequenceiq.cloudbreak.service.secret.domain.RotationSecret;
+import com.sequenceiq.cloudbreak.service.secret.service.SecretService;
+import com.sequenceiq.cloudbreak.service.stack.StackDtoService;
+
+@Component
+public class CMUserRotationExecutor implements RotationExecutor<CMUserRotationContext> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CMUserRotationExecutor.class);
+
+    @Inject
+    private ClusterApiConnectors clusterApiConnectors;
+
+    @Inject
+    private StackDtoService stackService;
+
+    @Inject
+    private SecretService secretService;
+
+    @Override
+    public void rotate(CMUserRotationContext rotationContext) {
+        LOGGER.info("Starting rotation of CM user for resource {} by creating a new user.", rotationContext.getResourceCrn());
+        try {
+            RotationSecret userRotationSecret = secretService.getRotation(rotationContext.getUserSecret());
+            RotationSecret passwordRotationSecret = secretService.getRotation(rotationContext.getPasswordSecret());
+            if (userRotationSecret.isRotation() && passwordRotationSecret.isRotation()) {
+                ClusterSecurityService clusterSecurityService = getClusterSecurityService(rotationContext);
+                String clientUser = secretService.get(rotationContext.getClientUserSecret());
+                String clientPassword = secretService.get(rotationContext.getClientPasswordSecret());
+                clusterSecurityService.createNewUser(
+                        userRotationSecret.getBackupSecret(),
+                        userRotationSecret.getSecret(),
+                        passwordRotationSecret.getSecret(),
+                        clientUser,
+                        clientPassword);
+            } else {
+                throw new SecretRotationException("User or password is not in rotation state in Vault, thus failure is expected at this point.", getType());
```

Comment:
I think this is a bit misleading, I would not expect failure :) Maybe something like this would be better: "thus rotation of CM user is not possible"